nixos/modules/services/finance/odoo.nix

```diff
@@ -113,7 +115,9 @@ in
           "HOME=%S/odoo"
           "ODOO_RC=${cfgFile}"
         ];
-      };
+      } // (lib.optionalAttrs cfg.autoInit {
+        ExecStartPre = "${cfg.package}/bin/odoo --init=INIT --database=odoo --db_user=odoo --stop-after-init";
```

Comment:
I'm pessimistic that all init steps are idempotent, such that running this before each start won't mess up a database with runtime data.

Might I suggest something like:

```bash
INITIALIZED="${cfg.data_dir}/.odoo.initialized"
if [ ! -e "$INITIALIZED" ]; then
  "${cfg.package}/bin/odoo --init=INIT --database=odoo --db_user=odoo --stop-after-init"
  touch "$INITIALIZED"
fi
```

Reply:
:+1: It seems to work, but better safe than sorry.

Reply:
I assume, if plugins are changed init does not need to be re-run? Then :+1: